HACKING.md

```diff
@@ -0,0 +1,87 @@
+# Hacking
+
+## Quickstart
+
+Use [ghcup][ghcup] to install the `cabal` cli tool and the ghc version we're using:
+
+```sh
+$ ghcup install-cabal
+$ ghcup install 8.8
+$ ghcup set 8.8
+```
+
+### Building
+
+In the base directory, run `cabal build`
+
+### Running tests
+
+In the base directory, run `cabal test`
+
+## Tools
+
+| name | description |
+| ---- | ----------- |
+| [ghcup][ghcup] | Used to manage installed versions of ghc and cabal-install |
+| ghc | The haskell compiler (installed via ghcup) |
+| cabal-install | The package manager we use (installed via ghcup) |
+| [haskell-language-server][hls] | LSP server for haskell projects (see below for installation instructions). Includes a formatter |
+| hlint | A linting + hints tool for haskell code. It provides really useful suggestions |
+
+### Installing haskell-language-server
+
+```sh
+git clone --recursive https://github.com/haskell/haskell-language-server.git
+cd haskell-language-server
+cabal update
+cabal install --installdir=$HOME/.cabal/bin --overwrite-policy=always
+```
+
+- VSCode: Install Install the "Haskell Language Server" plugin in VSCode. In the
+settings there's a `Language Server Haskell: Hie Variant` option, which you can
+set to `haskell-language-server`
+
+### Installing hlint
+
+```sh
+cabal update
+cabal install hlint --installdir=$HOME/.cabal/bin --overwrite-policy=always
+```
+
+- Run it with `hlint path/to/dir/or/file`
```

Comment:
```suggestion
- Run it with `hlint path/to/dir/or/file`
- VSCode: Install the `haskell-linter` plugin in vscode (`hoovercj.haskell-linter`).  It will
run automatically as you save, highlighting errors in your editor.
```